src/JsonSchema/Uri/Retrievers/FileGetContents.php

```diff
@@ -49,11 +49,10 @@ public function retrieve($uri)
         }
 
         $this->messageBody = $response;
-        if (!empty($http_response_header)) {
-            // $http_response_header cannot be tested, because it's defined in the method's local scope
-            // See http://php.net/manual/en/reserved.variables.httpresponseheader.php for more info.
-            $this->fetchContentType($http_response_header); // @codeCoverageIgnore
-        } else {                                            // @codeCoverageIgnore
+        $headers = strpos($uri, 'http') === 0 ? get_headers($uri) : array();
```

Comment:
It seems bad to me to switch to get_headers as, as far as I understand, this does a second HTTP request just to fetch the headers. Whereas $http_response_header is populated by file_get_contents in a single HTTP request.

So I'm not exactly sure what this fixes.

Reply:
Ok reading the issue again, I see what needs solving is fetchContentType needs to stop doing the early return. You need to iterate through all the headers and get the last content-type header's value, as that will be the final response. See https://github.com/composer/composer/blob/main/src/Composer/Util/RemoteFilesystem.php#L173-L177 for example.

Reply:
I see your points and only realise now that indeed a secondary call is made. This triggered me to look further into the issue and found an override for `file_get_contest` was made in the test causing the `$http_response_headers` to not be set in the tests. Further looking the test that depended on the override were having different behaviour between the test and runtime as `file_get_contents` can return false without invoking the defined error handler. Removing the test and the override seemed like to correct way forward.